File: infrastructure/db-core/src/main/java/com/nmh/commerce/utils/DurationConverter.java
package com.nmh.commerce.utils;

import jakarta.persistence.AttributeConverter;
import jakarta.persistence.Converter;

import java.time.Duration;

@Converter(autoApply = true)
public class DurationConverter implements AttributeConverter<Duration, String> {
    @Override
    public String convertToDatabaseColumn(Duration duration) {
        if(duration == null) {
            return null;
        }
        return duration.toString();
    }

    @Override
    public Duration convertToEntityAttribute(String s) {
        if(s == null) {
            return null;
        }
        return Duration.parse(s);
    }
}

File: infrastructure/db-core/src/main/kotlin/com/nmh/commerce/utils/DurationConverter.kt
package com.nmh.commerce.utils

import jakarta.persistence.AttributeConverter
import jakarta.persistence.Converter
import java.time.Duration

@Converter(autoApply = true)
class DurationConverter : AttributeConverter<Duration?, String?> {
    override fun convertToDatabaseColumn(duration: Duration?): String? {
        if (duration == null) {
            return null
        }
        return duration.toString()
    }

    override fun convertToEntityAttribute(s: String?): Duration? {
        if (s == null) {
            return null
        }
        return Duration.parse(s)
    }
}
